nightly-build.yml

```diff
@@ -0,0 +1,145 @@
+schedules:
+  - cron: "0 0 * * *"
+    displayName: Daily Midnight AzCopy Build
+    branches:
+      include:
+        - dev
+    always: true
+
+jobs:
+  - job: Nightly
+    # allow maximum build time, in case we have build congestion
+    timeoutInMinutes: 360
+    # Creating strategies for GOOS: Windows Server 2019 /macOS X Mojave 10.14/Ubuntu 20.04
+    strategy:
+      matrix:
+        Ubuntu-20:
+          imageName: 'ubuntu-20.04'
+          type: 'linux'
+        Windows:
+          imageName: 'windows-2019'
+          type: 'windows'
+        MacOS:
+          imageName: 'macOS-10.15'
+          type: 'mac-os'
+    pool:
+      vmImage: $(imageName)
+    variables:
+      - group: AzCopySecrets
+
+    steps:
+      - task: GoTool@0
+        inputs:
+          version: '1.15'
+      - script: |
+          go build -o "$(Build.ArtifactStagingDirectory)/azcopy_linux_amd64"
+          go build -tags "se_integration" -o "$(Build.ArtifactStagingDirectory)/azcopy_linux_se_amd64"
+          cp NOTICE.txt $(Build.ArtifactStagingDirectory)
+        displayName: 'Build linux'
+        condition: eq(variables.type, 'linux')
+
+      - script: |
+          go build -o "$(Build.ArtifactStagingDirectory)/azcopy_windows_amd64.exe"
+          go build -tags "windows_x86" -o "$(Build.ArtifactStagingDirectory)/azcopy_windows_386.exe"
+          cp NOTICE.txt $(Build.ArtifactStagingDirectory)
+        displayName: 'Build Windows'
+        condition: eq(variables.type, 'windows')
+
+      - script: |
+          go build -o "$(Build.ArtifactStagingDirectory)/azcopy_darwin_amd64"
+        displayName: 'Build MacOS'
+        condition: eq(variables.type, 'mac-os')
+
+      - task: DownloadSecureFile@1
+        name: ciGCSServiceAccountKey
+        displayName: 'Download GCS Service Account Key'
+        inputs:
+          secureFile: 'ci-gcs-dev.json'
+
+#      # Running Unit Tests on MacOS
```

Comment:
remove commented parts

Reply:
I promise I'll do it in next week of quality. I may need to refer it so I haven't done it.

Reply:
Perhaps it'd be best to store them to the side somewhere safe like OneNote for right now. @zezha-msft?